src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,144 @@
+    private static final Map<String, String> passwordsByLogin = Map.of(
+            "correctLogin", "correctPassword",
+            "testLoginSecond", "testPasswordSecond"
+    );
+    private static final List<String> shortLogins = List.of(
+            "", "t", "tt", "ttt", "tttt", "ttttt");
+    private static final List<String> shortPasswords = List.of(
+            "", "p", "pp", "ppp", "pppp", "ppppp");
+
+    private static StorageDao storageDao;
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void initVariables() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUser() {
+        user = new User();
+        user.setLogin(CORRECT_LOGIN);
+        user.setPassword(passwordsByLogin.get(CORRECT_LOGIN));
+        user.setAge(MIN_CORRECT_AGE);
+    }
+
+    @Test
+    void register_newUser_Ok() {
+        User actualUser = registrationService.register(user);
+
+        assertEquals(user, actualUser);
+        assertEquals(EXPECTED_ID, actualUser.getId());
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        runAssertThrows(null);
+    }
+
+    @Test
+    void register_loginPresentInStorage_notOk() {
+        user.setLogin(TEST_LOGIN_SECOND);
+        storageDao.add(user);
+
+        runAssertThrows(user);
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+
+        runAssertThrows(user);
+    }
+
+    @Test
+    void register_loginLengthLessThen6_notOk() {
+        for (String shortLogin : shortLogins) {
+            user.setLogin(shortLogin);
+
+            runAssertThrows(user);
+        }
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+
+        runAssertThrows(user);
+    }
+
+    @Test
+    void register_passwordLengthLessThen6_notOk() {
+        for (String shortPassword : shortPasswords) {
+            user.setPassword(shortPassword);
+
+            runAssertThrows(user);
+        }
+    }
+
+    @Test
+    void register_ageLessThen0Or0_notOk() {
+        for (int i = MAX_INCORRECT_AGE; i >= MIN_INCORRECT_AGE; i--) {
+            user.setAge(i);
+
+            runAssertThrows(user);
+        }
+    }
+
+    @Test
+    void register_ageLessThen18_notOk() {
+        for (int i = MIN_AGE_LESS_18; i < MIN_CORRECT_AGE; i++) {
+            user.setAge(i);
+
+            runAssertThrows(user);
+        }
+    }
+
+    @Test
+    void register_ageOver18_Ok() {
+        for (int i = MIN_CORRECT_AGE; i <= MAX_CORRECT_TESTING_AGE; i++) {
+            setUser();
+            user.setLogin(TEST_LOGIN_FOR_AGE + i);
+            user.setAge(i);
+
+            runAssertDoesNotThrow(user);
+        }
+    }
+
+    private void runAssertDoesNotThrow(User user) {
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }

```

Comment:
1. Separate method for single usage doesn't makes big sense
2. Test will fail anyway if exception is thrown and we do not expect it with assert
```suggestion
    @Test
    void register_ageOver18_Ok() {
        for (int i = MIN_CORRECT_AGE; i <= MAX_CORRECT_TESTING_AGE; i++) {
            setUser();
            user.setLogin(TEST_LOGIN_FOR_AGE + i);
            user.setAge(i);

            registrationService.register(user);
        }
    }
```